Lib/asyncio/taskgroups.py

```diff
@@ -158,10 +158,10 @@ def create_task(self, coro, *, name=None, context=None):
         if self._aborting:
             raise RuntimeError(f"TaskGroup {self!r} is shutting down")
         if context is None:
-            task = self._loop.create_task(coro)
+            task = self._loop.create_task(coro, name=name)
         else:
-            task = self._loop.create_task(coro, context=context)
-        task.set_name(name)
```

Comment:
Hm. If you're keeping it in the other place it should be kept here too, shouldn't it? It's exactly the same pattern -- call `loop.create_task(coro, name=name [, context=context])`.

Reply:
Sorry I think I'm confusing myself --

The call order for `create_task` from the `asyncio` module function would be (when the task factory is set):
```
asyncio.create_task(coro, name, context)
└── BaseEventLoop.create_task(coro, name, context)
    ├── BaseEventLoop._task_factory(coro, context)
    └── Task.set_name(name)
```

Similarly, for `TaskGroup`
```
TaskGroup.create_task(coro, name, context)
└── BaseEventLoop.create_task(coro, name, context)
    ├── BaseEventLoop._task_factory(coro, context)
    └── Task.set_name(name)
```

I think you we right from the onset -- we can remove the redundant call for both `asyncio.create_task` as well as `TaskGroup.create_task` since the [BaseEventLoop has the call when the task factory is set](https://github.com/python/cpython/blob/main/Lib/asyncio/base_events.py#L468), and [uses the `asyncio.Task` constructor with the name kwarg when the factory is not set](https://github.com/python/cpython/blob/main/Lib/asyncio/base_events.py#L458)